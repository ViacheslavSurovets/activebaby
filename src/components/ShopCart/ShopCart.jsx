import React from 'react';
import { ShopCartIcon } from '@components';
import styled from 'styled-components';
import { desktop, tablet, mobile } from '@utils/media';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

const CartWrapper = styled.figure`
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  background: ${ ( { theme } ) => theme.header.shopCartBackground };
  width: 100%;
  height: 100%;
  &:hover{
    background: ${ ( { theme } ) => theme.header.cartBackground }; 
      svg {
        fill: ${ ( { theme } ) => theme.header.cartColor };
      };
      span {
        color: ${ ( { theme } ) => theme.header.cartColor };
      }
      div {
       opacity: 1;
       visibility: visible;
       transition: .4s all .15s;
      }
 }
`;

const Counter = styled.span`
  ${ tablet },
  ${ mobile }{
    display: none;
  }
  ${ desktop }{
    color:${ ( { theme } ) => theme.white };
    margin-left: .5rem;
  }
`;

const ShopCartInfo = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  bottom: 0;
  right:0;
  transform: translateY(100%);
  width: 25rem;
  height: 10rem;
  font-size: 2rem;
  background: ${ ( { theme } ) => theme.header.cartBackground };
  border-bottom: 1px solid ${ ( { theme } ) => theme.header.bordersDropdown };
  border-left: 1px solid ${ ( { theme } ) => theme.header.bordersDropdown };
  color:${ ( { theme } ) => theme.header.linkColor };
  opacity: 0;
  visibility: hidden;
  transition: .4s all .15s;
  z-index: 1;
`;


const ShopCart = ( { count = 0 } ) => {
  const { t, i18n } = useTranslation ();
  return (
    <CartWrapper>
      <ShopCartIcon />
      <Counter>{ count }</Counter>
      <ShopCartInfo>
        <span>{ t ( 'header.headerCart.cartIsEmpty' ) }</span>
      </ShopCartInfo>
    </CartWrapper>
  );
};

export default ShopCart;

ShopCart.propTypes = {
  count: PropTypes.number
};
