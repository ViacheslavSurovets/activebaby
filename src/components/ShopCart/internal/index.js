export { ShopCartDropdownComponent } from './ShopCartDropDown';
export { CartItem } from './CartItem';
