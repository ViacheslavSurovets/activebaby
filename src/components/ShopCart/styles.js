import styled from 'styled-components';
import { desktop, mobile, tablet } from '@utils/media';

export const CartWrapper = styled.figure`
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  background: ${ ( { theme } ) => theme.header.shopCartBackground };
  width: 100%;
  height: 100%;
  &:hover{
    background: ${ ( { theme } ) => theme.header.cartBackground }; 
      svg {
        fill: ${ ( { theme } ) => theme.header.cartColor };
      };
      span {
        color: ${ ( { theme } ) => theme.header.cartColor };
      }
      //div {
      // opacity: 1;
      // visibility: visible;
      // transition: .4s all .15s;
      //}
 }
`;

export const Counter = styled.span`
  ${ tablet },
  ${ mobile }{
    display: none;
  }
  ${ desktop }{
    color:${ ( { theme } ) => theme.white };
    margin-left: .5rem;
  }
`;

// const styleCartDropdownTransition = css`
//    opacity: 1;
//    visibility: visible;
//    transition: 5s all .15s;
// `;

// const getTransition = ( props ) => {
//   if ( props.transition ) {
//     return styleCartDropdownTransition;
//   }
//   return;
// };

export const ShopCartDropdown = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  position: absolute;
  bottom: 0;
  right:0;
  transform: translateY(100%);
  width: 30rem;
  min-height: 20rem;
  font-size: 2rem;
  cursor:initial;
  background: ${ ( { theme } ) => theme.header.cartBackground };
  border-bottom: 1px solid ${ ( { theme } ) => theme.header.bordersDropdown };
  border-left: 1px solid ${ ( { theme } ) => theme.header.bordersDropdown };
  color:${ ( { theme } ) => theme.header.linkColor };
  z-index: 3;
  //opacity: 0;
  //visibility: hidden;
  //transition: .4s all .15s;
`;
