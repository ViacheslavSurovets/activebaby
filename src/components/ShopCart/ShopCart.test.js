import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import ShopCart from './ShopCart';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('ShopCart', ()=> {
  it('ShopCart Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <ShopCart  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
