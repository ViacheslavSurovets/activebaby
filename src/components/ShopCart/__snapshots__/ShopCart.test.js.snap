// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopCart ShopCart Snapshot 1`] = `ShallowWrapper {}`;
